File: intel-hex-to-bin.java/src/test/java/demo/HexTest.java
package demo;

import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Unit tests for {@link Hex}
 */
public class HexTest {

    @Test
    public void numberTo8BitHexStringTest() {
        assertEquals(Hex.to8BitHexString((short) 0x00), "0x00");
        assertEquals(Hex.to8BitHexString((short) 0x01), "0x01");

        assertEquals(Hex.to8BitHexString((short) 0x12), "0x12");
        assertEquals(Hex.to8BitHexString((short) 0xAB), "0xAB");
        assertEquals(Hex.to8BitHexString((short) 0xFF), "0xFF");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void numberTo8BitHexStringUnderflowTest() {
        Hex.to8BitHexString((short) -1);
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void numberTo8BitHexStringOverflowTest() {
        Hex.to8BitHexString((short) 0x100);
    }

    @Test
    public void numberTo16BitHexStringTest() {
        assertEquals(Hex.to16BitHexString(0x0000), "0x0000");
        assertEquals(Hex.to16BitHexString(0x0001), "0x0001");

        assertEquals(Hex.to16BitHexString(0x0012), "0x0012");
        assertEquals(Hex.to16BitHexString(0x00AB), "0x00AB");

        assertEquals(Hex.to16BitHexString(0x0123), "0x0123");
        assertEquals(Hex.to16BitHexString(0x0ABC), "0x0ABC");

        assertEquals(Hex.to16BitHexString(0x1234), "0x1234");
        assertEquals(Hex.to16BitHexString(0xABCD), "0xABCD");
        assertEquals(Hex.to16BitHexString(0xFFFF), "0xFFFF");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void numberTo16BitHexStringUnderflowTest() {
        Hex.to16BitHexString(-1);
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void numberTo16BitHexStringOverflowTest() {
        Hex.to16BitHexString(0x10000);
    }

    @Test
    public void arrayTo8BitHexStringTest() {
        assertEquals(Hex.to8BitHexString(new short[]{}), "[]");
        assertEquals(Hex.to8BitHexString(new short[]{0x12}), "[0x12]");
        assertEquals(Hex.to8BitHexString(new short[]{0x12, 0xAB}), "[0x12,0xAB]");
    }

}

File: intel-hex-to-bin.kotlin/src/test/kotlin/demo/HexTest.kt
package demo

import org.testng.Assert.assertEquals
import org.testng.annotations.Test

/**
 * Unit tests for [Hex]
 */
class HexTest {

    @Test
    fun numberTo8BitHexStringTest() {
        assertEquals(0x00.toShort().to8BitHexString(), "0x00")
        assertEquals(0x01.toShort().to8BitHexString(), "0x01")

        assertEquals(0x12.toShort().to8BitHexString(), "0x12")
        assertEquals(0xAB.toShort().to8BitHexString(), "0xAB")
        assertEquals(0xFF.toShort().to8BitHexString(), "0xFF")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun numberTo8BitHexStringUnderflowTest() {
        (0 - 1).toShort().to8BitHexString()
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun numberTo8BitHexStringOverflowTest() {
        0x100.toShort().to8BitHexString()
    }

    @Test
    fun numberTo16BitHexStringTest() {
        assertEquals(0x0000.to16BitHexString(), "0x0000")
        assertEquals(0x0001.to16BitHexString(), "0x0001")

        assertEquals(0x0012.to16BitHexString(), "0x0012")
        assertEquals(0x00AB.to16BitHexString(), "0x00AB")

        assertEquals(0x0123.to16BitHexString(), "0x0123")
        assertEquals(0x0ABC.to16BitHexString(), "0x0ABC")

        assertEquals(0x1234.to16BitHexString(), "0x1234")
        assertEquals(0xABCD.to16BitHexString(), "0xABCD")
        assertEquals(0xFFFF.to16BitHexString(), "0xFFFF")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun numberTo16BitHexStringUnderflowTest() {
        (0 - 1).to16BitHexString()
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun numberTo16BitHexStringOverflowTest() {
        0x10000.to16BitHexString()
    }

    @Test
    fun arrayTo8BitHexStringTest() {
        assertEquals(shortArrayOf().to8BitHexString(), "[]")
        assertEquals(shortArrayOf(0x12).to8BitHexString(), "[0x12]")
        assertEquals(shortArrayOf(0x12, 0xAB).to8BitHexString(), "[0x12,0xAB]")
    }

}
